portal-web/docroot/html/taglib/aui/button/end.jsp

```diff
@@ -21,7 +21,7 @@
 </c:if>
 
 <c:choose>
-	<c:when test="<%= Validator.isNotNull(escapedHREF) %>">
+	<c:when test='<%= Validator.isNotNull(escapedHREF) && !type.equals("cancel") %>'>
```

Comment:
The code here is fine, but the logic seems wrong. I don't think there is anything special about the 'cancel' button, that should affect if it should be a button or a link. I am aware that the ticket names the cancel button, but isn't that just the place where the bug was reported? What if we have a non-cancel button with `onClick` set? I'm sure there are many such cases in DXP. We would still want it to render it as a button and not a link, wouldn't we?

Could this be something like
```jsp
Validator.isNotNull(escapedHREF) && Validator.isNull(onClick)
```
or would this cause too many regressions? :thinking:

Reply:
Oh sorry, I misunderstood the intent, this is not about `onClick` attribute. But my question still remains: what is so special about the 'cancel' type? Why not have **all** buttons navigate?
```jsp
<aui:button href="#"> => <button onClick="redirect-to-href">
```

Reply:
Sometimes we need `<a>` as an output from `<aui:button>`. Cancel is also used for the message inside the button, and it's in many places of the Portal, if we had to change it everywhere we should upgrade the tag with a new attribute `output = a | button` but I think we don't have the time for it in this ticket

Reply:
What is the case where we need `<a>` as an output from `<aui:button>`? I understand we allowed it historically, but that was almost certainly a mistake. The current spree of accessibility changes is the proof, where something that looks like a button is expected to be a button and follow button-specific semantics and accessibility recommendations. Sidenote: we have the `<aui:a>` tag, if devs explicitly want a link.

Reply:
cc @dsanz

Reply:
I'd say yes, for now we want this only on the cancel buttons which are currently implemented as links. Fortunately we can check the `type` to limit the scope of the change. Still, as this may be a bit risky, we can merge this first and then see the impact of changing the rest of the cases where a link is generated when <aui:button> is used.

Reply:
Fair enough. I guess it makes sense from the risk management perspective.